libs/cart/src/resolvers/empty-cart-resolver.service.spec.ts

```diff
@@ -0,0 +1,89 @@
+import { provideMockActions } from '@ngrx/effects/testing';
+import { RouterTestingModule } from '@angular/router/testing';
+import { Router } from '@angular/router';
+
+import { DaffCart, DaffCartReducersState, daffCartReducers, DaffCartLoadSuccess }  from '@daffodil/cart';
+import { DaffCartFactory, DaffCartItemFactory } from '@daffodil/cart/testing';
+
+import { DaffEmptyCartResolver } from './empty-cart-resolver.service';
+import { DaffEmptyCartResolverRedirectUrl } from './tokens/empty-cart-resolver-redirect.token';
+import { DaffEmptyCartResolverModule } from './empty-cart-resolver.module';
+
+describe('DaffEmptyCartResolver', () => {
+  const actions$: Observable<any> = null;
+  let emptyCartResolver: DaffEmptyCartResolver;
+  let store: Store<DaffCartReducersState>;
+  let cartFactory: DaffCartFactory;
+  let cartItemFactory: DaffCartItemFactory;
+  let stubCart: DaffCart;
+	let router: Router;
+	const stubUrl = '/cart';
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        StoreModule.forRoot({
+          cart: combineReducers(daffCartReducers),
+        }),
+				RouterTestingModule,
+				DaffEmptyCartResolverModule
+      ],
+      providers: [
+				provideMockActions(() => actions$),
+				{ provide: DaffEmptyCartResolverRedirectUrl, useValue: stubUrl }
+      ]
+    });
+
+    emptyCartResolver = TestBed.get(DaffEmptyCartResolver);
+    cartFactory = TestBed.get(DaffCartFactory);
+    cartItemFactory = TestBed.get(DaffCartItemFactory);
+    stubCart = cartFactory.create();
+    store = TestBed.get(Store);
+    router = TestBed.get(Router);
+
+    spyOn(router, 'navigateByUrl');
+  }));
+
+  it('should be created', () => {
+    expect(emptyCartResolver).toBeTruthy();
+  });
+
+  describe('resolve', () => {
+
+    describe('when ResolverCartSuccessAction is dispatched with a cart', () => {
+      
+      it('should resolve with a DaffCartLoadSuccess action', () => {
+        emptyCartResolver.resolve().subscribe((resolvedValue) => {
+          expect(resolvedValue).toEqual(new DaffCartLoadSuccess(stubCart));
+				});
+
+				store.dispatch(new DaffCartLoadSuccess(stubCart));
+      });
+
+      describe('and cart is empty', () => {
+        
+        it('should redirect to the provided DaffEmptyCartRedirectUrl', () => {
+					stubCart.items = [];
+
+          emptyCartResolver.resolve().subscribe();
+          store.dispatch(new DaffCartLoadSuccess(stubCart));
+					expect(router.navigateByUrl).toHaveBeenCalledWith(stubUrl);
+        });
+      });
+
+      describe('and cart is not empty', () => {
+				
+        it('should not redirect to the provided DaffEmptyCartRedirectUrl', () => {
+					stubCart = cartFactory.create({items: cartItemFactory.create()});
+
+          emptyCartResolver.resolve().subscribe();
+          store.dispatch(new DaffCartLoadSuccess(stubCart));
+					expect(router.navigateByUrl).not.toHaveBeenCalledWith(stubUrl);
+        });
+      });
```

Comment:
No, it would only redirect to the `DaffCartResolverRedirectUrl` if the action is not a DaffCartLoadSuccess. So it would need be a DaffCartLoadFailure, a DaffCartStorageFailure, or a...I can't remember the third option...in order to redirect to the `DaffCartResolverRedirectUrl`. And this is intended behavior, because otherwise, the page will just not redirect to anywhere and the user will be left with a blank page.

Reply:
oh you're right. I didn't look at this closely enough.